README.md
# Sathiyakugan - Personal portfolio



changelog.md
# Portfolio Jekyll Theme - Change Log



assets_home/js/custom-js.js
$(window).on('load', function() {
    /*=========================================================================
       Isotope
       =========================================================================*/

    var $container = $('.portfolio-wrapper');
    $container.imagesLoaded( function() {
        $('.portfolio-wrapper').isotope({
            // options
            itemSelector: '[class*="col-"]',
            percentPosition: true,
            masonry: {
                // use element for option
                columnWidth: '[class*="col-"]'
            }
        });
    });

    $('.portfolio-filter').on( 'click', 'li', function() {
        var filterValue = $(this).attr('data-filter');
        $container.isotope({ filter: filterValue });
    });

    // change is-checked class on buttons
    $('.portfolio-filter').each( function( i, buttonGroup ) {
        var $buttonGroup = $( buttonGroup );
        $buttonGroup.on( 'click', 'li', function() {
            $buttonGroup.find('.current').removeClass('current');
            $( this ).addClass('current');
        });
    });

    var filterValue = $('#portfolio-filter-first').attr('data-filter');
    $container.isotope({ filter: filterValue });


    $('.pop-box-model').on("click", function () {
        var theGoodStuff = $(this).find('.mfp-hide');
        $.magnificPopup.open({
            items: {
                src: theGoodStuff,
            },
            type: 'inline',
            fixedContentPos: true,
            fixedBgPos: true,
            overflowY: 'auto',
            closeBtnInside: false,
            preloader: false,
            midClick: true,
            removalDelay: 300,
            mainClass: 'my-mfp-zoom-in'
        });
    });


    // var bolbyPopup = function(){
    //     /*=========================================================================
    //             Magnific Popup
    //     =========================================================================*/
    //     $('.work-image').magnificPopup({
    //         type: 'image',
    //         closeBtnInside: false,
    //         mainClass: 'my-mfp-zoom-in',
    //     });
    //
    //     $('.work-content').magnificPopup({
    //         type: 'inline',
    //         fixedContentPos: true,
    //         fixedBgPos: true,
    //         overflowY: 'auto',
    //         closeBtnInside: false,
    //         preloader: false,
    //         midClick: true,
    //         removalDelay: 300,
    //         mainClass: 'my-mfp-zoom-in'
    //     });
    //
    //     $('.work-video').magnificPopup({
    //         type: 'iframe',
    //         closeBtnInside: false,
    //         iframe: {
    //             markup: '<div class="mfp-iframe-scaler">'+
    //                 '<div class="mfp-close"></div>'+
    //                 '<iframe class="mfp-iframe" frameborder="0" allowfullscreen></iframe>'+
    //                 '</div>',
    //
    //             patterns: {
    //                 youtube: {
    //                     index: 'youtube.com/',
    //
    //                     id: 'v=',
    //
    //                     src: 'https://www.youtube.com/embed/%id%?autoplay=1'
    //                 },
    //                 vimeo: {
    //                     index: 'vimeo.com/',
    //                     id: '/',
    //                     src: '//player.vimeo.com/video/%id%?autoplay=1'
    //                 },
    //                 gmaps: {
    //                     index: '//maps.google.',
    //                     src: '%id%&output=embed'
    //                 }
    //
    //             },
    //
    //             srcAction: 'iframe_src',
    //         }
    //     });
    //
    //     $('.gallery-link').on('click', function () {
    //         $(this).next().magnificPopup('open');
    //     });
    //
    //     $('.gallery').each(function () {
    //         $(this).magnificPopup({
    //             delegate: 'a',
    //             type: 'image',
    //             closeBtnInside: false,
    //             gallery: {
    //                 enabled: true,
    //                 navigateByImgClick: true
    //             },
    //             fixedContentPos: false,
    //             mainClass: 'my-mfp-zoom-in',
    //         });
    //     });
    // }
    //
    // bolbyPopup();



    /*=========================================================================
     Infinite Scroll
     =========================================================================*/
    var curPage = 1;
    var pagesNum = $(".portfolio-pagination").find("li a:last").text();   // Number of pages

    $container.infinitescroll({
            itemSelector: '.grid-item',
            nextSelector: '.portfolio-pagination li a',
            navSelector: '.portfolio-pagination',
            extraScrollPx: 0,
            bufferPx: 0,
            maxPage: 6,
            loading: {
                finishedMsg: "No more works",
                msgText: '',
                speed: 'slow',
                selector: '.load-more',
            }
        },
        // trigger Masonry as a callback
        function( newElements ) {

            var $newElems = $( newElements );
            $newElems.imagesLoaded(function(){
                $newElems.animate({ opacity: 1 });
                $container.isotope( 'appended', $newElems );
            });

            bolbyPopup();

            // Check last page
            curPage++;
            if(curPage == pagesNum) {
                $( '.load-more' ).remove();
            }

        });

    $container.infinitescroll( 'unbind' );

    $( '.load-more .btn' ).on('click', function() {
        $container.infinitescroll( 'retrieve' );
        // display loading icon
        $( '.load-more .btn i' ).css('display', 'inline-block');
        $( '.load-more .btn i' ).addClass('fa-spin');

        $(document).ajaxStop(function () {
            setTimeout(function(){
                // hide loading icon
                $( '.load-more .btn i' ).hide();
            }, 1000);
        });
        return false;
    });

    /* ======= Mobile Filter ======= */

// bind filter on select change
    $('.portfolio-filter-mobile').on( 'change', function() {
        // get filter value from option value
        var filterValue = this.value;
        // use filterFn if matches value
        filterValue = filterFns[ filterValue ] || filterValue;
        $container.isotope({ filter: filterValue });
    });

    var filterFns = {
        // show if number is greater than 50
        numberGreaterThan50: function() {
            var number = $(this).find('.number').text();
            return parseInt( number, 10 ) > 50;
        },
        // show if name ends with -ium
        ium: function() {
            var name = $(this).find('.name').text();
            return name.match( /ium$/ );
        }
    };
});


assets_home/js/custom-index.js

// Your web app's Firebase configuration
var firebaseConfig = {
    apiKey: "AIzaSyAxbdJjXl2JYcuVTiRKd7lIcZc5I2HeBK8",
    authDomain: "portfolio-f0afb.firebaseapp.com",
    databaseURL: "https://portfolio-f0afb.firebaseio.com",
    projectId: "portfolio-f0afb",
    storageBucket: "portfolio-f0afb.appspot.com",
    messagingSenderId: "548137266756",
    appId: "1:548137266756:web:6050a446fb37217d3cd611",
    measurementId: "G-SX0ZBPEEL1"
};
// Initialize Firebase
firebase.initializeApp(firebaseConfig);
firebase.analytics();


$( document ).ready(function() {
    firebase.database().ref('/user/').once('value').then(function(snapshot) {
        setUserProfile(snapshot.val())
    });
    // firebase.database().ref('/social-media/').once('value').then(function(snapshot) {
    //     setSocialProfile(snapshot.val())
    // });


    //grab a form
    const form = document.querySelector('#contact-me-form');

    //push on form submit
    if (form) {
        //grab an input
        const contactMeName = form.querySelector('#contact-me-name');
        const contactMeEmail = form.querySelector('#contact-me-email');
        const contactMeSubject = form.querySelector('#contact-me-subject');
        const contactMeMessage= form.querySelector('#contact-me-message');

        form.addEventListener('submit', function (evt) {
            evt.preventDefault();
            firebasePush(contactMeName,contactMeEmail,contactMeSubject,contactMeMessage);

            //shows alert if everything went well.
            bootstrap_alert_contactme()
            form.reset();
        })
    }


    //grab a form
    const alertBarform = document.querySelector('#alertbar-form');

    //push on form submit
    if (alertBarform) {
        alertBarform.addEventListener('submit', function (evt) {
            evt.preventDefault();
            const alertBarEmail = alertBarform.querySelector('#alertbar-email');
            firebasePushAlertEmail(alertBarEmail);
            //shows alert if everything went well.
            bootstrap_alert_Subscribe();
            alertBarform.reset();
        })
    }


});

function setUserProfile(user) {

    if ($('#user-name')){
        $('#user-name').html(user["name_with_caps"])
    }

    if ( $('#user-title')){$('#user-title').html(user["title"])
       }

    if ($('#user-description')){
        $('#user-description').html(user["description"])
    }
}


function bootstrap_alert_contactme () {
    $('#contact-me-alert_placeholder').append(`<div class="alert alert-success alert-dismissible">
                  <button type="button" class="close" data-dismiss="alert">&times;</button>
                  <strong>Success!</strong> Thanks for contacting me. I will get back to you shortly.
                </div>`);
    alertTimeout(3000); //Called here
}

function bootstrap_alert_Subscribe () {
    $('#contact-me-alert_placeholder').append(`<div class="alert alert-success alert-dismissible" style="margin-top: 10px">
                  <button type="button" class="close" data-dismiss="alert">&times;</button>
                  <strong>Success!</strong> Thanks for subscribing to our newsletter.
                </div>`);
    alertTimeout(2000); //Called here
}

function alertTimeout(wait){
    setTimeout(function(){
        $('#contact-me-alert_placeholder').children('.alert:first-child').fadeOut("slow",function() {
            // After animation completed:
            $( this ).remove();
        })
    }, wait);
}

//create a functions to push
function firebasePush(contactMeName,contactMeEmail,contactMeSubject,contactMeMessage) {
    //prevents from braking
    if (!firebase.apps.length) {
        firebase.initializeApp(config);
    }

    //push itself
    var mailsRef = firebase.database().ref('Contact Me Emails').push().set(
        {
            name: contactMeName.value,
            email: contactMeEmail.value,
            subject: contactMeSubject.value,
            message: contactMeMessage.value,
            date: (new Date()).toLocaleString()
        }
    );

}

//create a functions to push
function firebasePushAlertEmail(contactMeEmail) {
    //prevents from braking
    if (!firebase.apps.length) {
        firebase.initializeApp(config);
    }

    //push itself
    var mailsRef = firebase.database().ref('Subscribe Me Emails').push().set(
        {
            email: contactMeEmail.value,
            date: (new Date()).toLocaleString()
        }
    );

}





// function setSocialProfile(user) {
//
//     $('#user-github').attr('href', (user["github"]));
//     $('#user-linkedin').attr('href', (user["linkedin"]));
//     $('#user-stack-overflow').attr('href', user["stack-overflow"]);
//     $('#user-twitter').attr('href', user["twitter"]);
// }

assets_home/js/main.js
/*
===================================
          Table Content
===================================
1.  responsive menu slideing
2.  code for mobile menu 
3.  menu options custom affix
4.  isotope
5.  skill circle progress bar
6.  progress bar
7.  Swiper Slider
    - portfolio-slider
    - testimonial slider
    - blog slider
    - blog slider for home-page-two
    - post-img-slider
    - more-case-slider
    - test-slider
    -
8. Scroll to top
*/


(function($){
    "use strict";   

    // responsive menu slideing
    if ($(window).width() < 992) {
      $(".navbar-collapse>ul>li>a, .navbar-collapse ul.sub-menu>li>a").on("click", function() {
        var element = $(this).parent("li");
        if (element.hasClass("open")) {
          element.removeClass("open");
          element.find("li").removeClass("open");
          element.find("ul").slideUp(500,"linear");
        }
        else {
          element.addClass("open");
          element.children("ul").slideDown();
          element.siblings("li").children("ul").slideUp();
          element.siblings("li").removeClass("open");
          element.siblings("li").find("li").removeClass("open");
          element.siblings("li").find("ul").slideUp();
        }
      });
      
    }
    
  //js code for mobile menu 
    $(".menu-toggle").on("click", function() {
        $(this).toggleClass("is-active");
    });

    //jQuery for page scrolling feature - requires jQuery Easing plugin

    $("ul.menu").each( function( i, liList ) {
      var $liList = $( liList );

      $liList.on( "click", "li a.scroll-smoth", function(event) {
        var $anchor = $(this);
        $("html, body").stop().animate({
            scrollTop: $($anchor.attr("href")).offset().top - 70
        }, 1500, "easeInOutExpo");
        event.preventDefault();
      });
    });

    //menu options custom affix
    var fixed_top = $(".header-section");
    $(window).on("scroll", function(){
      
      if( $(this).scrollTop() > 50 ){  
        fixed_top.addClass("animated fadeInDown menu-fixed");
      }
      else{
        fixed_top.removeClass("animated fadeInDown menu-fixed");
      }
    });


    $(window).on("load", function() {
        
      // isotope
      $(".grid").isotope({
        // options
        itemSelector: ".filter-item"
      });
      var $gallery = $(".grid").isotope({
        // options
      });

      // filter items on button click
      $(".portfolio-item-button").on( "click", "span", function() {
          var filterValue = $(this).attr("data-filter");
          $gallery.isotope({ filter: filterValue });

      });

      $(".portfolio-item-button").on( "click", "span", function() {
          $(this).addClass("active").siblings().removeClass("active");
      });
    });

    $("a[data-rel^=lightcase]").lightcase();

    // skill circle progress bar
    var skillLevel1 = $(".skill-item .first").data("skill-level");
    var skillLevel2 = $(".skill-item .second").data("skill-level");
    var skillLevel3 = $(".skill-item .third").data("skill-level");
    var skillLevel4 = $(".skill-item .fourth").data("skill-level");
    //first.circle
    $(".first.circle").circleProgress({
      value: "0." + skillLevel1,
      size: 110,
      emptyFill: "transparent",
      fill: {
        color: "#39b54a"
      }
    }).on("circle-animation-progress", function(event, progress) {
      $(this).find("strong").html(Math.round(95 * progress) + "<span>%</span>");
    });

    //second.circle
    $(".second.circle").circleProgress({
      value: "0." + skillLevel2,
      size: 110,
      emptyFill: "transparent",
      fill: {
        color: "#f26522"
      }
    }).on("circle-animation-progress", function(event, progress) {
      $(this).find("strong").html(Math.round(skillLevel2 * progress) + "<i>%</i>");
    });

    //third.circle
    $(".third.circle").circleProgress({
      value: "0." + skillLevel3,
      size: 110,
      emptyFill: "transparent",
      fill: {
        color: "#6e41ff"
      }
    }).on("circle-animation-progress", function(event, progress) {
      $(this).find("strong").html(Math.round(70 * progress) + "<span>%</span>");
    });

    //fourth.circle
    $(".fourth.circle").circleProgress({
      value: "0." + skillLevel4,
      size: 110,
      emptyFill: "transparent",
      fill: {
        color: "#ec008c"
      }
    }).on("circle-animation-progress", function(event, progress) {
      $(this).find("strong").html(Math.round(90 * progress) + "<span>%</span>");
    });

     // progress bar
    $(".progressbar").each(function(){
      $(this).find(".bar").animate({
        "width": $(this).attr("data-perc")
      },3000);
      $(this).find(".label").animate({
        "left": $(this).attr("data-perc")
      },3000);
    });

    //portfolio-slider
    var swiper = new Swiper(".portfolio-slider", {
      slidesPerView: 2,
      slidesPerColumn: 2,
      spaceBetween: 20,
      pagination: {
        el: ".swiper-pagination",
        clickable: true,
      },
    });

    // testimonial slider
    var swiper = new Swiper(".testimonial-slider", {
      effect: "coverflow",
      loop: 0,
      slidesPerView: "auto",
      grabCursor: 1,
      parallax: 0,
      centeredSlides: true,
      coverflowEffect: {
          rotate: 0,
          stretch: 80,
          depth: 200,
          modifier: 1,
          slideShadows: !1
      },
      pagination: {
        el: ".swiper-pagination",
        dynamicBullets: true,
      },
      navigation: {
        nextEl: ".button-next",
        prevEl: ".button-prev",
      },
    });

    // blog slider
    var swiper = new Swiper(".blog-slider", {
      slidesPerView: 2,
      spaceBetween: 0,
      pagination: {
        el: ".swiper-pagination",
        clickable: true,
      },
      breakpoints: {
        1920: {
          slidesPerView: 2,
          spaceBetween: 0,
        },
        1440: {
          slidesPerView: 1,
          spaceBetween: 0,
        },
        1199: {
          slidesPerView: 2,
          spaceBetween: 0,
        },
        575: {
          slidesPerView: 1,
          spaceBetween: 0,
        }
      }
    });

    // blog slider for home-page-two
    var swiper = new Swiper(".blog-slider-two", {
      slidesPerView: 3,
      spaceBetween: 0,
      pagination: {
        el: ".swiper-pagination",
        clickable: true,
      },
      breakpoints: {
        991: {
          slidesPerView: 2,
          spaceBetween: 0,
        },
        575: {
          slidesPerView: 1,
          spaceBetween: 0,
        }
      }
    });

    // post-img-slider
    var swiper = new Swiper(".post-img-slider", {
      slidesPerView: 1,
      spaceBetween: 0,
      navigation: {
        nextEl: ".swiper-button-next",
        prevEl: ".swiper-button-prev",
      },
      autoplay: {
        delay: 1000,
      },
    });

    // more-case-slider
    var swiper = new Swiper(".more-case-slider", {
      pagination: {
        el: ".swiper-pagination",
      },
    });

    // test-slider
    var swiper = new Swiper(".testi-slider", {
      slidesPerView: 1,
      spaceBetween: 0,
    });

    $(".work-smaple-main").slick({
      slidesToShow: 1,
      slidesToScroll: 1,
      arrows: false,
      fade: true,
      asNavFor: ".work-smaple-nav"
    });
    $(".work-smaple-nav").slick({
      slidesToShow: 4,
      slidesToScroll: 1,
      asNavFor: ".work-smaple-main",
      dots: true,
      centerMode: true,
      focusOnSelect: true,
      prevArrow: $(".prev"),
      nextArrow: $(".next"),
    });

    // Show or hide the sticky footer button
    $(window).on("scroll", function() {
      if ($(this).scrollTop() > 200) {
          $(".scroll-to-top").fadeIn(200);
      } else {
          $(".scroll-to-top").fadeOut(200);
      }
    });

    // Animate the scroll to top
    $(".scroll-to-top").on("click", function(event) {
        event.preventDefault();
        $("html, body").animate({scrollTop: 0}, 300);
    });
  
  })(jQuery); 